### PR TITLE
Instrumentation

### PR DESCRIPTION
**Pull request by fletna**

### CODE REVIEW

File: coda-prototype/editorUI.js [hunk 1]
```diff
@@ -62,6 +62,17 @@ var codeEditorManager =  {
 
             $("#word-textarea").find(".tag").css({"background-color": event.color.toHex()});
 
+            // update the activity stack
+            /* TODO find a solution that doesn't flood events
```
Comment (lukechurch) [inline coda-prototype/editorUI.js]: Could you file an issue for this?

File: coda-prototype/editorUI.js [hunk 2]
```diff
@@ -291,9 +331,14 @@ var codeEditorManager =  {
         $(".add-code-row").on("click", function() {
             let newCode = addCodeInputRow("","", "#ffffff", "", []); // todo will return codeObject
             codeEditorManager.updateCodePanel(newCode);
-        });
-
 
+            // update the activity stack
+            storage.saveActivity({
+                "category": "SCHEME",
+                "message": "Added new code " + newCode.id + " to scheme " + tempScheme.id,
```
Comment (lukechurch) [inline coda-prototype/editorUI.js]: See comment below, this would be hard to machine parse

File: coda-prototype/editorUI.js [hunk 3]
```diff
@@ -400,17 +445,28 @@ var codeEditorManager =  {
             wordTextarea.find(".tag").css({'background-color': color});
             codeObj.addWords(event.item);
             regexField.val(regexMatcher.generateOrRegex(codeObj["words"]));
-
+            // update the activity stack
+            storage.saveActivity({
+                "category": "SCHEME",
+                "message": "Added word " + event.item + " to code " + codeObj.id + " in scheme " + tempScheme.id,
```
Comment (lukechurch) [inline coda-prototype/editorUI.js]: In general I'd consider making these messages more machine parse able. Something like the psuedo-code  
```"Added word to code: " + {"word" : event.item, "code": codeObj.id}.toJSON()```

File: coda-prototype/editorUI.js [hunk 4]
```diff
@@ -400,17 +445,28 @@ var codeEditorManager =  {
...
         });
+
         $(selectObj).on('itemRemoved', function(event) {
             // event.item: contains the item
             codeObj.deleteWords([event.item]);
             regexField.val(regexMatcher.generateOrRegex(codeObj["words"]));
+            // update the activity stack
+            storage.saveActivity({
+                "category": "SCHEME",
+                "message": "Deleted word " + event.item + " from code " + codeObj.id + " in scheme " + tempScheme.id,
```
Comment (lukechurch) [inline coda-prototype/editorUI.js]: As above

File: coda-prototype/editorUI.js [hunk 5]
```diff
@@ -545,19 +610,29 @@ var codeEditorManager =  {
         let schemeOrderIndex = messageViewerManager.codeSchemeOrder.indexOf(schemeId);
         if (schemeOrderIndex > -1) messageViewerManager.codeSchemeOrder.splice(schemeOrderIndex, 1);
 
-        if (Object.keys(schemes).length == 0) {
+        if (Object.keys(newDataset.schemes).length == 0) {
             // create new default coding scheme
             let newScheme = new CodeScheme(UIUtils.randomId([]), "default", true);
             newScheme.codes.set(newScheme.id + "-" + "1", new Code(newScheme, newScheme.id + "-" + "1", "test", "#ffffff", "", false));
-            schemes[newScheme.id] = newScheme;
+            //schemes[newScheme.id] = newScheme;
```
Comment (lukechurch) [inline coda-prototype/editorUI.js]: Why is this line commented out? Should it be deleted?

File: coda-prototype/editorUI.js [hunk 6]
```diff
@@ -545,19 +610,29 @@ var codeEditorManager =  {
...
             newDataset.schemes[newScheme.id] = newScheme;
             messageViewerManager.addNewSchemeColumn(newScheme);
-            messageViewerManager.codeSchemeOrder.push(newScheme.id);
+            messageViewerManager.codeSchemeOrder.push(newScheme.id + "");
```
Comment (lukechurch) [inline coda-prototype/editorUI.js]: What does adding an empty string to newScheme.id do?

File: coda-prototype/scrollbar.js [hunk 7]
```diff
@@ -266,7 +279,7 @@ var scrollbarManager = {
             pagesToLoad = pagesToLoad-2;
         }
 
-        messageViewerManager.lastLoadedPageIndex = [];
+        messageViewerManager.lastLoadedPageIndex = []; // todo what is this?
```
Comment (lukechurch) [inline coda-prototype/scrollbar.js]: Please can you add an issue?

File: coda-prototype/ui.ts [hunk 8]
```diff
@@ -114,7 +114,7 @@ class SchemaEntry {
 }
 
 // Services 
-
+/*
```
Comment (lukechurch) [inline coda-prototype/ui.ts]: Either remove, or add note explaining what the commented out code does?

File: coda-prototype/editorUI.js (hunk 1 above)
Comment (fletna) [inline coda-prototype/editorUI.js]: Now filed in #62

**Pull request merged**